src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -252,27 +252,31 @@ private static IResourceBuilder<ProjectResource> WithProjectDefaults(this IResou
         var isHttp2ConfiguredInAppSettings = config["Kestrel:EndpointDefaults:Protocols"] == "Http2";
         var adjustTransport = (EndpointAnnotation e) => e.Transport = isHttp2ConfiguredInAppSettings ? "http2" : e.Transport;
 
-        if (builder.ApplicationBuilder.ExecutionContext.IsRunMode)
+        foreach (var schemeGroup in kestrelEndpointsByScheme)
         {
-            foreach (var schemeGroup in kestrelEndpointsByScheme)
-            {
-                // If there is only one endpoint for a given scheme, we use the scheme as the endpoint name
-                // Otherwise, we use the actual endpoint names from the config
-                var schemeAsEndpointName = schemeGroup.Count() <= 1 ? schemeGroup.Key : null;
+            // If there is only one endpoint for a given scheme, we use the scheme as the endpoint name
+            // Otherwise, we use the actual endpoint names from the config
+            var schemeAsEndpointName = schemeGroup.Count() <= 1 ? schemeGroup.Key : null;
 
-                foreach (var endpoint in schemeGroup)
+            foreach (var endpoint in schemeGroup)
+            {
+                builder.WithEndpoint(schemeAsEndpointName ?? endpoint.EndpointName, e =>
                 {
-                    builder.WithEndpoint(schemeAsEndpointName ?? endpoint.EndpointName, e =>
-                    {
-                        e.Port = endpoint.BindingAddress.Port;
-                        e.UriScheme = endpoint.BindingAddress.Scheme;
-                        e.IsProxied = false; // turn off the proxy, as we cannot easily override Kestrel bindings
-                        e.Transport = adjustTransport(e);
-                    },
-                    createIfNotExists: true);
-                }
+                    e.Port = endpoint.BindingAddress.Port;
```

Comment:
This is an issue for publish mode, we only want to set the TargetPort not the Port. Settings a port does not work for the default http and https endpoint as those are 80 and 443.

Reply:
Hmmm, I'm not crazy about having this difference between Publish and Run.

Stepping back, how is this Kestrel situation different from having a user manually define an external endpoint with some hard coded `Port`? It's going to generate the same manifest, and it's going to blow up on ACA.

It feels quirky to have the Kestrel ports treated as external as runtime, and as a target at Publish time. Instead, can we do this:
- There is exactly one Kestrel endpoint that is used for external traffic. Probably the one named `http`
- We always treat that port as a **target port**, both at runtime and Publish
- Aspire comes up with a proper external facing Port for it:
  - At runtime it can be a random port, unless the user somehow tells us what to use
  - At publish time, we always map both http 80 and https 443 to it in the manifest.

Reply:
Ok so that's not going to work, because in local Run mode, you can't have replicas if you set the target port. So we have two very different worlds:

Local run:
- Pros: Can have many external ports
- Cons: Target ports must be generated to support proxies and replicas, since they must be unique across the box
Result: the user can pick the Port but not the Target Port

ACA:
- Cons: Can only have one http and one https port
- Pros: Target ports are scoped within the container, so all the replicas can use the same one
Result: the user can pick the Target Port but not the Port